api/tests/mocha/services/extension-libs.js

```diff
@@ -0,0 +1,128 @@
+const sinon = require('sinon');
+const chai = require('chai');
+
+const service = require('../../../src/services/extension-libs');
+const db = require('../../../src/db');
+
+describe.only('Extension Libs service', () => {
+
+  let dbGet;
+
+  beforeEach(() => {
+    dbGet = sinon.stub(db.medic, 'get');
+  });
+  afterEach(() => sinon.restore());
+
+  describe('isLibChange', () => {
+
+    it('false if invalid params', () => {
+      const actual = service.isLibChange();
+      chai.expect(actual).to.be.false;
+    });
+
+    it('false if different doc', () => {
+      const actual = service.isLibChange({ id: 'myform' });
+      chai.expect(actual).to.be.false;
+    });
+
+    it('true if extension lib doc', () => {
+      const actual = service.isLibChange({ id: 'extension-libs' });
+      chai.expect(actual).to.be.true;
+    });
+
+  });
+
+  describe('getAll', () => {
+
+    it('handles 404', async () => {
+      dbGet.throws({ status: 404 });
+      const actual = await service.getAll();
+      chai.expect(actual.length).to.eq(0);
+    });
+
+    it('throws anything else', async () => {
+      dbGet.throws({ status: 403 });
+      try {
+        await service.getAll();
+      } catch(e) {
+        chai.expect(e.status).to.equal(403);
+        return;
+      }
+      throw new Error('Expected error to be thrown');
```

Comment:
I think I like my way better. They both correctly fail the test which is the important thing, but when you use `chai.expect.fail` it throws an Error under the hood, which gets caught by the `catch(e)`. Then it fails with a hard to diagnose message because it's actually failing on the `chai.expect` line... eg:

Using `chai.expect.fail`:

```
  1) Extension Libs service
       get
         throws anything else:
     AssertionError: expected undefined to equal 403
      at Context.<anonymous> (api/tests/mocha/services/extension-libs.spec.js:109:34)
```

Using `throw new Error` outside of the `catch`:

```
  1) Extension Libs service
       get
         throws anything else:
     Error: Expected error to be thrown
      at Context.<anonymous> (api/tests/mocha/services/extension-libs.spec.js:112:13)
```

Reply:
Ohhh yeah that is better failure behavior!  The whole setup, though, still feels primitive....  I was reading a little bit and with `chai-as-promised`, we can do this:

```js
      await chai.expect(service.getAll()).to.eventually.be.rejected
        .with.property('status', 403);
```

That seems to give good failure errors both in the case where the expected status is not found and when the promise is not rejected.  

I am fine leaving things how you have it, but I am going to try and use `chai-as-promised` more in the future....